src/lib/reasoners/ite_rel.ml

```diff
@@ -54,21 +54,37 @@ module TB =
         if c <> 0 then c else Stdlib.compare b1 b2
     end)
 
-type t =
-  { pending_deds      : Ex.t ME2.t;
-    guarded_pos_deds  : SE2.t ME.t;
-    guarded_neg_deds  : SE2.t ME.t;
-    assumed_pos_preds : Ex.t ME.t;
-    assumed_neg_preds : Ex.t ME.t;
-  }
-
-let empty _ =
-  { pending_deds  = ME2.empty;
-    guarded_pos_deds  = ME.empty;
-    guarded_neg_deds  = ME.empty;
-    assumed_pos_preds = ME.empty;
-    assumed_neg_preds = ME.empty;
-  }
+(* The present theory simplifies the ite t of the form
+    ite(pred, t1, t2)
+   where pred is an assumed predicate by adding the equation
+   t = t1 or t = t2 according to the truth value of pred. *)
+
+type t = {
+  pending_deds      : Ex.t ME2.t;
+  (* Map of pending deducations to their explanation. A deduction is just
```

Comment:
What does "pending" mean? My intuition is that this a deduction that is true under the corresponding explanation, and the explanation is assumed to be true in the current context, is that accurate?

Reply:
This is a documentation PR, it should not change any names.

From what you are saying, I understand that a "pending deduction" is a deduction that is true in the current model, associated with the explanation that made it true? If so we should add the info to the doc of `pending_deds` (saying that "`pending_deds` is the map of pending deductions to their explanations" is not very useful to understand the code if we don't explain what a "pending deduction" is).